tests/helpers.py

```diff
@@ -0,0 +1,389 @@
+
+set_exe_marker('acs')
+use_calacs = pytest.mark.skipif(not HAS_CALXXX['acs'], reason='no CALACS')
+
+set_exe_marker('wfc3')
+use_calwf3 = pytest.mark.skipif(not HAS_CALXXX['wfc3'], reason='no CALWF3')
+
+set_exe_marker('stis')
+use_calstis = pytest.mark.skipif(not HAS_CALXXX['stis'], reason='no CALSTIS')
+
+
+def _download_file(url, filename, filemode='wb', timeout=None):
+    """Generic remote data download."""
+    if url.startswith('http'):
+        r = requests.get(url, timeout=timeout)
+        with open(filename, filemode) as fout:
+            fout.write(r.content)
+    elif url.startswith('ftp'):  # TODO: Support filemode and timeout.
+        urllib.request.urlretrieve(url, filename=filename)
+    else:  # pragma: no cover
+        raise ValueError('Unsupported protocol for {}'.format(url))
+
+
+def download_crds(refdir, refname, timeout=None):
+    """Download a CRDS file from HTTP or FTP to current directory."""
+    # CRDS file for given name never changes, so no need to re-download.
+    if os.path.exists(refname):
+        return
+
+    try:
+        url = 'http://ssb.stsci.edu/cdbs/{}/{}'.format(refdir, refname)
+        _download_file(url, refname, timeout=timeout)
+    except Exception:  # Fall back to FTP
+        url = 'ftp://ftp.stsci.edu/cdbs/{}/{}'.format(refdir, refname)
+        _download_file(url, refname, timeout=timeout)
+
+
+def download_file_cgi(tree, project, filename, filemode='wb', timeout=30,
+                      allow_remote_ref=False):
+    """
+    Download remote data to current directory from Central Storage
+    or using CGI script interface.
+
+    Parameters
+    ----------
+    tree : {'rt', 'rtx', 'null'}
+        Test tree:
+
+        * rt = dev
+        * rtx = public
+        * null = neither, grab first match (e.g., for STAK)
+
+    project : str
+        Path containing data in the test tree.
+        For example, ``hstcal/acs/calacs_e``.
+
+    filename : str
+        Filename of the data.
+
+    filemode : str
+        Mode of the output file writer. By default, it saves as binary
+        and overwrites existing file.
+
+    timeout : number or `None`
+        This is not a time limit on the entire response download;
+        rather, an exception is raised if the server has not issued
+        a response for timeout seconds (more precisely, if no bytes
+        have been received on the underlying socket for timeout seconds).
+        If no timeout is specified explicitly, requests do not time out.
+
+    allow_remote_ref : bool
+        If set to `True`, instead of copying or downloading to current
+        directory, return the full path. This should only be used for
+        "truth" images.
+
+    """
+    # NOTE: This could be explicitly controlled using pytest fixture
+    #       but too many ways to do the same thing would be confusing.
+    #       Refine this logic if using pytest fixture.
+    cs_file = os.path.join('/eng/ssb2/tests', tree, project, filename)
```

Comment:
@mdlpstsci , the "prepend" part is here. However, if you change this for one test, you change it for all the test. You currently cannot change this in a per-test basis.